src/write_sensu.c

```diff
@@ -39,6 +39,53 @@
 #include <stddef.h>
 
 #include <stdlib.h>
+#ifdef HAVE_KERNEL_SOLARIS
```

Comment:
Shouldn't this be `KERNEL_SOLARIS`? Otherwise the substitute is not used. When I change the line accordingly then I get a compilation error of the following form.
The problem is that the machine where I compile the package has `vasprintf` but I disallow the linker to bind against the libc contract that provides it.
```
gmake[3]: Entering directory '/home/dam/mgar/pkg/collectd/trunk/work/solaris10-sparc/build-isa-sparcv8plus/collectd-5.5.2/src'
  CC       write_sensu.lo
write_sensu.c:52:12: error: static declaration of 'vasprintf' follows non-static declaration
 static int vasprintf(char **str, const char *fmt, va_list args) {
            ^
In file included from ./daemon/collectd.h:34:0,
                 from write_sensu.c:29:
/usr/include/stdio.h:263:12: note: previous declaration of 'vasprintf' was here
 extern int vasprintf(char **, const char *, __va_list);
            ^
write_sensu.c:76:12: error: static declaration of 'asprintf' follows non-static declaration
 static int asprintf(char **str, const char *fmt, ...) {
            ^
In file included from ./daemon/collectd.h:34:0,
                 from write_sensu.c:29:
/usr/include/stdio.h:261:12: note: previous declaration of 'asprintf' was here
 extern int asprintf(char **, const char *, ...);
            ^
Makefile:3937: recipe for target 'write_sensu.lo' failed
gmake[3]: *** [write_sensu.lo] Error 1
```